arches_controlled_lists/utils/skos.py

```diff
@@ -0,0 +1,204 @@
+                list_id = self.generate_uuidv5_from_subject(baseuuid, scheme)
+                new_list = List(list_id)
+
+                for predicate, object in graph.predicate_objects(subject=scheme):
+                    # Get List name from a ConceptScheme's title element
+                    if predicate == DCTERMS.title:
+
+                        if not self.language_exists(object, allowed_languages):
+                            for lang in models.Language.objects.all():
+                                allowed_languages[lang.code] = lang
+
+                        val = self.unwrapJsonLiteral(object)
+                        new_list.name = val["value"]
+
+                self.lists[scheme] = new_list
+
+            # Create lookups for valuetypes used during Concept processing
+            value_types = models.DValueType.objects.all()
+            skos_value_types = value_types.filter(
+                Q(namespace="skos") | Q(namespace="arches")
+            )
+            skos_note_and_label_types = skos_value_types.filter(
+                Q(category="note") | Q(category="label")
+            )
+            skos_value_types = {
+                valuetype.valuetype: valuetype for valuetype in skos_value_types
+            }
+            skos_note_and_label_types = {
+                valuetype.valuetype: valuetype
+                for valuetype in skos_note_and_label_types
+            }
+
+            ### List items & values ###
+            # Concepts become ListItems & ListItemValues
+            for concept, v, o in graph.triples((None, RDF.type, SKOS.Concept)):
+                list_item_id = self.generate_uuidv5_from_subject(baseuuid, concept)
+                list_item = ListItem(id=list_item_id)
+
+                # rdf:about is fallback URI for a concept, unless it has dcterms:identifier
+                uri = self.unwrapJsonLiteral(str(concept))
+                sortorder = 0
+
+                for predicate, object in graph.predicate_objects(subject=concept):
+                    if predicate == DCTERMS.identifier:
+                        uri = self.unwrapJsonLiteral(str(object))["value"]
+
+                    elif predicate == SKOS.inScheme:
+                        list_item.list = self.lists[object]
+
+                    elif any(
+                        type in predicate for type in skos_note_and_label_types.keys()
+                    ):
+                        if not self.language_exists(object, allowed_languages):
+                            for lang in models.Language.objects.all():
+                                allowed_languages[lang.code] = lang
+
+                        object_language = (
+                            allowed_languages[object.language] or default_lang
+                        )
+                        relation_or_value_type = predicate.replace(SKOS, "").replace(
+                            ARCHES, ""
+                        )
+                        list_item_value = ListItemValue(
```

Comment:
Same question about overwrites -- but since no pk provided here, it would just potentially create dupes, is that okay?